eunkyung/app.js

```diff
@@ -34,25 +34,146 @@ app.get('/ping', (req, res) => {
 })
 
 app.post('/sign_up', async (req, res, next) => {
-    const { name, password, email } = req.body
+    const { name, password, email, profileImage } = req.body
     await mysqlDataSource.query(
         `INSERT INTO users (
             name, 
             password, 
-            email
+            email,
+            profile_image
             ) 
             VALUES (
+                ?,
                 ?,
                 ?,
                 ?
                 );`,
-        [name, password, email]
+        [name, password, email, profileImage]
     )
     res.status(201).json({ message: 'userCreated' })
 })
 
 
 
+app.post('/posting', async (req, res) => {
+    const { title, content, postImageUrl, userId } = req.body
+    await mysqlDataSource.query(
+        `INSERT INTO posts (
+            title,
+            content,
+            post_image_url,
+            user_id
+        )
+        VALUES (
+            ?,
+            ?,
+            ?,
+            ?
+        );`,
+        [title, content, postImageUrl, userId]
+    )
+    res.status(201).json({ message: 'postCreated' })
+})
+
+app.get('/posts', async (req, res) => {
+    await mysqlDataSource.query(
+        `SELECT
+            users.id as userId,
+            users.profile_image as userprofileImage,
+            posts.user_id as postingId,
+            posts.post_image_url as postingImageUrl,
+            posts.content as postingContent 
+        FROM users 
+        INNER JOIN posts ON users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+
+app.get('/posts/:userId', async (req, res) => {
+    const { userId } = req.params;
+
+    await mysqlDataSource.query(
+        `SELECT
+            users.id as userId,
+            users.profile_image as userprofileImage,
+            pp.postings
+            FROM users
+        LEFT JOIN (
+            SELECT 
+            user_id,
+            JSON_ARRAYAGG(
+                JSON_OBJECT(
+                    'postingId', posts.id,
+                    'postingImageUrl', posts.post_image_url,
+                    'postingContent', posts.content
+                ) 
+            ) 
+        as postings
+        FROM posts 
+        GROUP BY user_id  
+        ) pp ON users.id = pp.user_id
+            WHERE users.id = ${userId};`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+app.patch('/edit', async (req, res) => {
```

Comment:
edit 역시 동사형으로 엔드포인트 url에 사용되기에는 조금 부적합해보입니다.
```suggestion
app.patch('/posts', async (req, res) => {
```